text/0000-rustflags.md

```diff
@@ -0,0 +1,383 @@
+Rust developer to set the debuginfo level for the current crate and have full debugging symbols but still allow all other dependencies to
+be fully optimized. This will save on both compilation time and binary size.
+
+As with the previous example, being able to specify only saving the temporary files, (`-C save-temps`), for the current crate as opposed
+to all crates in the dependency graph. This will save on disk space by not having all of the unnecessary temp files being saved on each
+invocation of the Rust compiler.
+
+There are some unstable flags that would also benefit from this feature such as the `-Z unpretty=<val>` option. This could be used to
+expand the input source to allow for easier debugging of macros and proc macros. This will allow a single crate to have its input
+expanded and lessen the amount of source expanded which would allow for Rust developer to better understand how a macro is affecting
+their source code.
+
+## --rustflags
+
+The `--rustflags` option will allow a Rust developer to pass any set of rustc flags to the root crate being built. This will allow a
+simple command such as `cargo test` to have the option of setting the `-C instrument-coverage` flag for a single crate and run the unit
+tests ensuring coverage data is collected for this crate and this crate only. For example, let's take crate `foo`:
+
+Cargo.toml:
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+```
+
+cargo CLI (Output lines have been removed for simplicity):
+```
+cargo test --rustflags -C instrument-coverage ;
+
+Compiling foo v0.1.0 (...)
+     Running `rustc --crate-name foo --edition=2021 src/lib.rs --crate-type lib ... -C instrument-coverage`
+     Running `rustc --crate-name foo --edition=2021 src/lib.rs --test ... -C instrument-coverage`
+    Finished test [unoptimized + debuginfo] target(s) in 1.35s
+     Running `target/debug/deps/foo-669448d9b4043564`
+
+running 1 test
+test tests::it_works ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+```
+
+Running this command will build the crate `foo` with the flag `-C instrument-coverage` passed to the invocation of rustc
+for the crate `foo` only. Upstream dependencies would not be instrumented as well as the standard libraries, which is not the case
+now, saving on compilation time.
+
+## --rustflags for a workspace
+
+Cargo supports two types of [workspaces](https://doc.rust-lang.org/cargo/reference/workspaces.html), a workspace with a root package and
+a workspace with a virtual manifest, meaning a "primary" package does not exist.
+
+For workspaces that contain a root package, any `--rustflags <RUSTFLAGS>` options set will be passed to the invocation of the Rust
+compiler for the root crate only. If the `workspace.default-members` manifest key has been set, then only the members listed as a
+default member will have the `--rustflags` values passed to the invocation of rustc.
```

Comment:
I feel like tying the definition of "root package" to the primary package or the default-members doesn't seem like the right way to go.

Unfortunately, I'm not sure of anything else.  Maybe "whatever is 'selected' on the command-line" which will be inferred from the primary package or default-members but allows a user finer control (`cargo build --package foo` would make `foo` a root for that invocation)